src/lib/Libtpp/tpp_em.c

```diff
@@ -691,12 +753,28 @@ tpp_em_del_fd(void *em_ctx, int fd)
  *
  */
 int
-tpp_em_wait(void *em_ctx, em_event_t **ev_array, int timeout)
+tpp_em_pwait(void *em_ctx, em_event_t **ev_array, int timeout, const sigset_t *sigmask)
 {
 	poll_context_t *ctx = (poll_context_t *) em_ctx;
 	int nready;
 	int i;
 	int ev_count;
+#ifndef PBS_HAVE_PPOLL
+	sigset_t origmask;
+#endif
+
+#ifdef PBS_HAVE_PPOLL
+	nready = ppoll(ctx->fds, ctx->curr_nfds, timeout, sigmask);
+#else
+	if (sigmask) {
+		if (sigprocmask(SIG_SETMASK, sigmask, &origmask) == -1)
+			return -1;
+	}
+
+	if (sigmask) {
+		sigprocmask(SIG_SETMASK, &origmask, NULL);
+	}
+#endif
 
 	nready = poll(ctx->fds, ctx->curr_nfds, timeout);
```

Comment:
The poll() call should be in between the blocking and unblocking the signals using sigprocmask